### PR TITLE
Prepare for release 0.5.0

### PR DESCRIPTION
**Pull request by alculquicondor**
/assign @tenzen-y 

I created release notes manually by inspecting https://github.com/kubeflow/mpi-operator/compare/v0.4.0...79a9b379647cd1d9bcc540f97410cc6a4e94c8aa

Part-of: https://github.com/kubeflow/mpi-operator/issues/563

### CODE REVIEW

**Comment by alculquicondor** [comment]
/hold
@tenzen-y also wants to upgrade k8s libraries.

**Review by terrytangyuan — APPROVED** [review]
Thanks!

/lgtm

File: RELEASE.md
Comment (tenzen-y) [inline RELEASE.md:1]: LGTM!

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/kubeflow/mpi-operator
 
-go 1.20
+go 1.22
```
Comment (tenzen-y) [inline go.mod:3]: Could you submit the dedicated PR since all other PRs depend on this Go version update?

**Review by tenzen-y — COMMENTED** [review]
Basically, lgtm

File: RELEASE.md [hunk 2]
```diff
@@ -1,5 +1,17 @@
 # MPI Operator Releases
 
+## Release v0.5.0
```
Comment (tenzen-y) [inline RELEASE.md:3]: Could we mention that we have started to support K8s v1.28 and v1.29?

**Comment by tenzen-y** [comment]
Part-of: #563

File: RELEASE.md (hunk 2 above)
Comment (alculquicondor) [reply RELEASE.md:3]: uhm... it should have run before. But I can mention it in the cleanups.

Are we missing any PR before I rebase?

Comment (tenzen-y) [reply RELEASE.md:3]: > uhm... it should have run before. But I can mention it in the cleanups.

That makes sense.

> Are we missing any PR before I rebase?

The last piece is https://github.com/kubeflow/mpi-operator/pull/634.

**Comment by alculquicondor** [comment]
rebased

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm
/hold cancel
